velox/type/tz/TimeZoneMap.cpp

```diff
@@ -62,8 +71,16 @@ TTimeZoneDatabase buildTimeZoneDatabase(
     // Every single other time zone entry (outside of offsets) needs to be
     // available in external/date or this will throw.
     else {
-      timeZonePtr = std::make_unique<TimeZone>(
-          entry.second, entry.first, date::locate_zone(entry.second));
+      const date::time_zone* zone;
+      try {
+        zone = locateZoneExt(entry.second);
+      } catch (date::invalid_timezone& err) {
+        // Timezone not found in OS, skip it.
+        LOG(WARNING) << "Unable to load [" << entry.second
+                     << "] from local timezone database: " << err.what();
+        continue;
```

Comment:
@zhztheplayer You're right, in this case we leave the entire TimeZone pointers as null, so it will do the right thing. Thanks for clarifying. 

It would probably be worth it to add a small comment here explaining what in this case we continue the iteration and leave the pointer unchanged (nullptr), for the next unadvised reader like myself :)